draft-ietf-jsonpath-base.md

```diff
@@ -549,58 +547,30 @@ The result is a list containing `0`, `1`.
 This is the concatenation of three lists, two of length one containing
 `0`, `1`, respectively, and one of length zero.
 
-As a consequence of this approach, if any of the selectors selects no nodes,
-then the whole query selects no nodes.
+As a consequence of this approach, if any of the __APPENDERS__ produces an empty nodelist,
+then the whole query produces an empty nodelist.
 
-In what follows, the semantics of each selector are defined for each type
+In what follows, the semantics of each __APPENDER__ are defined for each type
 of node.
 
+## Root Identifier
 
-## Selectors
-
-A JSONPath query consists of a sequence of selectors. Valid selectors are
-
-  * Root selector `$` (used at the start of a query and in expressions)
-  * Dot selector `.<name>`, used with object member names exclusively
-  * Dot wildcard selector `.*`
-  * Index selector `[<index>]`, where `<index>` is either a (possibly
-    negative, see {{index-semantics}}) array index or an object member name
-  * Index wildcard selector `[*]`
-  * Array slice selector `[<start>:<end>:<step>]`, where the optional
-    values `<start>`, `<end>`, and `<step>` are integer literals
-  * List selector `[<sel1>,<sel2>,...,<selN>]`, holding a comma
-    separated list of index and slice selectors
-  * Filter selector `[?(<expr>)]`
-  * Current item selector `@` (used in expressions)
-  * Descendants selectors starting with a double dot `..`
-
-Note that processing the dot selector, string-valued index selector,
-and filter selector all potentially require matching strings against
-strings, with those strings coming from the JSONPath and from member
-names and string values in the JSON to which it is being applied.
-Two strings MUST be considered equal if and only if they are identical
-sequences of Unicode scalar values. In other words, normalization operations
-MUST NOT be applied to either the string from the JSONPath or from the JSON
-prior to comparison.
-
-### Root Selector
-
-#### Syntax
+### Syntax
 {: unnumbered}
 
-Every valid JSONPath query MUST begin with the root selector `$`.
+Every valid JSONPath query MUST begin with the root identifier `$`.
```

Comment:
What does the word "valid" mean here?
Do we also describe "invalid JSONPath queries"?

This sentence has the disadvantage that it makes @ queries into something different, which then require another word for $ queries and @ queries collectively.

Reply:
e96a561ef79ac2457c3d44f9bdb0fca77112ca2b deletes the word "valid".

I need to think more about $ and @ terminology.

Reply:
I'll keep this conversation open, but we can do this in the next round.